include/assimp/mesh.h

```diff
@@ -129,27 +129,24 @@ struct aiFace
     //! The maximum value for this member is #AI_MAX_FACE_INDICES.
     unsigned int mNumIndices;
 
-    //! Pointer to the indices array. Size of the array is given in numIndices.
-    unsigned int* mIndices;
```

Comment:
This will break every single Assimp client in existence, and require non-trivial changes when upgrading. Ideally this kind of optimization would preserve the public-facing struct API in some fashion, otherwise it'll probably have to wait until the next major version bump.

I took a stab at something similar a while ago in an attempt to minimize heap usage (https://github.com/assimp/assimp/compare/master...jdduke:noheap_face). It preserves the outward-facing aiFace struct data members, but eliminates heap allocations for faces with index count <= 4 (albeit in a somewhat crude fashion, index storage on the mesh is a better long term solution, but unless we make mIndices a pointer into the mesh index array, which has its own set of issues, this is a breaking change).